fix: Minor Navigational and Permission Rationale Dialog Issues

farhan-arshad-dev opened the pull request:
### Description

Jira Ticket: [LEARNER-10460](https://2u-internal.atlassian.net/browse/LEARNER-10460)

- Fix the navigation issue when the thread is unavailable.
- Removed unnecessary parameters passed on thread clicked.
- Place notification primer behind a feature flag `PUSH_NOTIFICATIONS_ENABLED`.
- Add category property in `Notification:Setting Permission Status` event.
- Fix permission Rationale Dialog Issue.
- Fix test cases.

Commit:
fix: navigation issue when thread is unavailable

- Resolved an issue where discussion notification navigation would stop on the threads screen if the thread was not found in the list.

fixes: LEARNER-10460

Commit:
fix: removed unnecessary parameters passed on thread clicked

fixes: LEARNER-10460

Commit:
fix: place notification primer behind a feature flag

fixes: LEARNER-10460

Commit:
chore: add category property in `Notification:Setting Permission Status`

HamzaIsrar12 commented on discussion/src/main/java/org/openedx/discussion/presentation/threads/DiscussionThreadsViewModel.kt:
```diff
@@ -207,6 +209,8 @@ class DiscussionThreadsViewModel(
 
     fun showNotificationsPrimer(context: Context, fm: FragmentManager) {
-        pushGlobalManager.showNotificationsPrimer(context, fm)
+        if (config.isPushNotificationsEnabled()) {
+            pushGlobalManager.showNotificationsPrimer(context, fm)
+        }
```
Maybe write this around `_showPrimer.emit(true)` so we have one less recompossion.

farhan-arshad-dev replied:
There is no recomposition since it is not being used as a State.

HamzaIsrar12 commented on app/src/main/java/org/openedx/app/MainViewModel.kt:
```diff
@@ -90,6 +90,7 @@ class MainViewModel(
         analytics.logEvent(event.eventName, buildMap {
             put(AppAnalyticsKey.NAME.key, event.biValue)
             put(AppAnalyticsKey.STATUS.key, permissionStatus.status)
+            put(AppAnalyticsKey.CATEGORY.key, AppAnalyticsKey.NOTIFICATIONS)
```
Can we move the `Notification:Setting Permission Status` analytic event in the `notifications` module?

farhan-arshad-dev replied:
addressed.

HamzaIsrar12 reviewed (changes requested):
Minor nits. The "Permission Rationale Dialog Issue" is still persisting. I've shared the steps to reproduce in the ticket.

Commit:
fix: Permission Rationale Dialog Issue

- Added a check for shouldShowRequestPermissionRationale in the Course Dashboard screen.
- Ensures that the rationale dialog is only shown when necessary.
- Prevents unnecessary permission prompts for a smoother user experience.

fixes: LEARNER-10460

Commit:
refactor: `Notification:Setting Permission Status` event in the notifications module

HamzaIsrar12 commented on course/src/main/java/org/openedx/course/presentation/container/CourseContainerViewModel.kt:
```diff
@@ -769,4 +772,14 @@ class CourseContainerViewModel(
+        activity: FragmentActivity,
+        permissionLauncher: ActivityResultLauncher<String>,
+    ) {
+        pushManager.requestNotificationPermission(
```
Won't this make it dependent on the Notifications Module, as we aren't providing any other implementation for it?

farhan-arshad-dev replied:
moved `PermissionUtils` to the core module.

Commit:
refactor: `PermissionUtils` to core module

HamzaIsrar12 reviewed (approved):
LGTM 🏎️ 

Minor nit: We need to revert the changes in **CourseContainerViewModel** and **CourseContainerViewModelTest**.

Commit:
refactor: remove unused code

k1rill reviewed (approved):
LGTM
